quickstarts/microsoft.network/internal-loadbalancer-create/main.bicep

```diff
@@ -0,0 +1,119 @@
+@description('address prefix')
+param vnetAddressPrefix string = '10.0.0.0/16'
+
+@description('Subnet prefix')
+param subnetPrefix string = '10.0.0.0/24'
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+var virtualNetworkName = 'myVNet'
+var subnetName = 'myBackendSubnet'
+var loadBalancerName = 'myLoadBalancer'
+var nicName = 'myNIC1'
+var lbSku = 'Standard'
+var lbFrontEndName = 'loadBalancerFrontEnd'
+var lbBackEndName = 'loadBalancerBackEnd'
+var lbRuleName = 'HTTPRule'
+var lbProbeName = 'TCPPort80HealthProbe'
+
+resource vnet 'Microsoft.Network/virtualNetworks@2021-05-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+  }
+}
+
+resource subnet 'Microsoft.Network/virtualNetworks/subnets@2021-05-01' = {
+  name: subnetName
+  parent: vnet
+  properties: {
+    addressPrefix: subnetPrefix
+  }
+}
+
+resource nic 'Microsoft.Network/networkInterfaces@2021-05-01' = {
+  name: nicName
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: subnet.id
+          }
+          loadBalancerBackendAddressPools: [
+            {
+              id: '${lb.id}/backendAddressPools/${lbBackEndName}' //resourceId('Microsoft.Network/loadBalancers/backendAddressPools', loadBalancerName, 'loadBalancerBackEnd')
```

Comment:
use resourceId here as you have below

Reply:
Updated, wasn't sure what was the best practice for this use case.